RuleEngine/Application.cs

```diff
@@ -38,27 +53,49 @@ public void Run(string[] args)
 				Location = new Location() { Name = "Test" }
 			};
 
-			double doubleResult = 0;
-			bool boolResult = false;
+			//double doubleResult = 0;
+			//bool boolResult = false;
 
 			foreach (var rule in rules)
 			{
-				if (string.IsNullOrWhiteSpace(rule.InputType))
-				{
-					doubleResult = _parser.Parse<double>(rule.Expression);
-					_log.LogInformation($"Result of expression {rule.Expression} is {doubleResult}");
+                var returnType = GetShortcutType(rule.ReturnType) ?? Type.GetType(rule.ReturnType, false, true);
+                var methods = typeof(IParser).GetMethods().Where(m => m.ContainsGenericParameters).ToList();
+
+                if (string.IsNullOrWhiteSpace(rule.InputType))
+                {
+                    var method = methods.First(
+                        m => m.ReturnType.IsGenericParameter && m.GetGenericArguments().Length == 1 &&
```

Comment:
Not sure this is the best way to find the generic method V Parse<V>(string expression)

Reply:
It's probably also not necessary - I would imagine 99% of the time, a bool return type would be desired for evaluating a truthy expression.  The "ReturnType" was just a thought on the BusinessRule object.